sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Integration.Tests/DistroWebAppLiveTests.cs

```diff
@@ -104,41 +117,79 @@ public async Task VerifyDistro()
-                query: $"AppRequests | where Url == '{TestServerUrl}' | where AppRoleName == '{RoleName}' | top 1 by TimeGenerated");
+                query: $"AppRequests | where Url == '{TestServerUrl}' | where AppRoleName == '{RoleName}' | top 1 by TimeGenerated",
+                expectedAppRequest: new ExpectedAppRequest
+                {
+                    Url = TestServerUrl,
+                    AppRoleName = RoleName,
+                });
 
-            await VerifyTelemetry(
+            await QueryAndVerifyMetric(
                 description: "Metric for outgoing request, from testhost",
-                query: $"AppMetrics | where Name == 'http.client.duration' | where AppRoleName == '{RoleName}' | where Properties.['net.peer.name'] == 'localhost' | top 1 by TimeGenerated");
+                query: $"AppMetrics | where Name == 'http.client.request.duration' | where AppRoleName == '{RoleName}' | where Properties.['server.address'] == 'localhost' | top 1 by TimeGenerated",
+                expectedAppMetric: new ExpectedAppMetric
+                {
+                    Name = "http.client.request.duration",
+                    AppRoleName = RoleName,
+                    Properties = new List<KeyValuePair<string, string>>
+                    {
+                        new("server.address", "localhost"),
+                    },
+                });
 
-            await VerifyTelemetry(
+            await QueryAndVerifyMetric(
                 description: "Metric for incoming request, from WebApp",
-                query: $"AppMetrics | where Name == 'http.server.duration' | where AppRoleName == '{RoleName}' | where Properties.['net.host.name'] == 'localhost' | top 1 by TimeGenerated");
+                query: $"AppMetrics | where Name == 'http.server.request.duration' | where AppRoleName == '{RoleName}' | top 1 by TimeGenerated",
+                expectedAppMetric: new ExpectedAppMetric
+                {
+                    Name = "http.server.request.duration",
+                    AppRoleName = RoleName,
+                    Properties = new(),
+                });
 
-            await VerifyTelemetry(
+            await QueryAndVerifyTrace(
                 description: "ILogger LogInformation, from WebApp",
-                query: $"AppTraces | where Message == '{LogMessage}' | where AppRoleName == '{RoleName}' | top 1 by TimeGenerated");
+                query: $"AppTraces | where Message == '{LogMessage}' | where AppRoleName == '{RoleName}' | top 1 by TimeGenerated",
+                expectedAppTrace: new ExpectedAppTrace
+                {
+                    Message = LogMessage,
+                    AppRoleName = RoleName,
+                });
         }
 
-        private async Task VerifyTelemetry(string description, string query)
+        private async Task QueryAndVerifyDependency(string description, string query, ExpectedAppDependency expectedAppDependency)
```

Comment:
All of these methods validates the same thing, as a follow up you could consider a generic method like this one.
```
private async Task QueryAndVerifyTelemetry<T>(string description, string query, T expectedTelemetry) where T : IExpectedAppTelemetry
{
    LogsTable? logsTable = await _logsQueryClient!.QueryTelemetryAsync(description, query);
    ValidateExpectedTelemetry(description, logsTable, expectedTelemetry);
}
```